routing/restriction_loader.cpp

```diff
@@ -15,13 +51,46 @@ RestrictionLoader::RestrictionLoader(MwmValue const & mwmValue)
     ReaderSource<FilesContainerR::TReader> src(*m_reader);
     m_header.Deserialize(src);
 
-    RestrictionSerializer::Deserialize(m_header, m_restrictions, src);
+    RestrictionVec restrictionsOnly;
+    RestrictionSerializer::Deserialize(m_header, m_restrictions /* restriction no */,
+                                       restrictionsOnly, src);
+    ConvertRestrictionsOnlyToNoAndSort(graph, restrictionsOnly, m_restrictions);
   }
   catch (Reader::OpenException const & e)
   {
     m_header.Reset();
     LOG(LERROR,
         ("File", m_countryFileName, "Error while reading", RESTRICTIONS_FILE_TAG, "section.", e.Msg()));
+    throw;
+  }
+}
+
+void ConvertRestrictionsOnlyToNoAndSort(IndexGraph const & graph,
+                                        RestrictionVec const & restrictionsOnly,
+                                        RestrictionVec & restrictionsNo)
+{
+  for (Restriction const & o : restrictionsOnly)
+  {
+    CHECK_EQUAL(o.m_featureIds.size(), 2, ("Only two link restrictions are support."));
+    if (!graph.IsRoad(o.m_featureIds[0]) || !graph.IsRoad(o.m_featureIds[1]))
+      continue;
+
+    // Looking for a joint of an intersection of |o| features.
+    Joint::Id const common =
+        GetCommonEndJoint(graph.GetRoad(o.m_featureIds[0]), graph.GetRoad(o.m_featureIds[1]));
+    if (common == Joint::kInvalidId)
```

Comment:
if featureIds[0] is from and featureIds[1] is to then why common is not a  featureIds[0].GetEndingJointId()?